src/middlewared/middlewared/plugins/cache.py

```diff
@@ -355,13 +356,15 @@ def get_uncached_user(self, username=None, uid=None):
             u = pwd.getpwuid(uid)
         else:
             return {}
+
         return {
             'pw_name': u.pw_name,
             'pw_uid': u.pw_uid,
             'pw_gid': u.pw_gid,
             'pw_gecos': u.pw_gecos,
             'pw_dir': u.pw_dir,
-            'pw_shell': u.pw_shell
+            'pw_shell': u.pw_shell,
+            'grouplist': os.getgrouplist(u.pw_name, u.pw_gid) if getgroups else []
```

Comment:
I don't think we should be sending an empty list in the response as that potentially makes the consumer think that the list is in reality empty which is not true. Maybe not add the key at all or send `null` for distinction ?

Reply:
Sure. I can omit if it's not specified.